WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetails.swift

```diff
@@ -54,6 +59,30 @@ struct CouponDetails: View {
         ]
     }
 
+    private var actionSheetButtons: [ActionSheet.Button] {
+        var buttons: [ActionSheet.Button] = [
+            .default(Text(Localization.copyCode), action: {
+                UIPasteboard.general.string = viewModel.couponCode
+                let notice = Notice(title: Localization.couponCopied, feedbackType: .success)
+                noticePresenter.enqueue(notice: notice)
+                ServiceLocator.analytics.track(.couponDetails, withProperties: ["action": "copied_code"])
+            }),
+            .default(Text(Localization.shareCoupon), action: {
+                showingShareSheet = true
+                ServiceLocator.analytics.track(.couponDetails, withProperties: ["action": "shared_code"])
+            }),
+        ]
+
+        if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.couponEditAndDelete) {
```

Comment:
I believe it could be better to pass the feature flag in the init method of the view model, and use just a variable from the view model here. In that way, we can also implement some tests since we can pass a mock `ServiceLocator` to the view model. BTW you will encounter some merge conflicts because I used this approach here https://github.com/woocommerce/woocommerce-ios/pull/6528 and I used instead a method for generating the action sheet buttons 👍 let me know if you need an hand with the merge 😉